appmap/flask.py

```diff
@@ -47,21 +48,14 @@ def request_params(req):
 
 class AppmapFlask(AppmapMiddleware):
     def __init__(self, app=None):
+        super().__init__()
         self.app = app
         if app is not None:
             self.init_app(app)
 
     def init_app(self, app):
-        if not Env.current.enabled:
```

Comment:
Removing all these checks means that even if recording is disabled (e.g. with `APPMAP=false`), the remote-recording routes will still be added to the app, right? Is there some reason that's necessary?

Reply:
That's right, the remote-recording routes will still be added.

Without removing this check some testcase wasn't working.  I think it was [test_starts_disabled](https://github.com/getappmap/appmap-python/blob/master/appmap/test/web_framework.py#L235)

I wish my commit history wasn't squashed into a single commit so that I could find through the commit comments which testcase this was.

Reply:
I don't think we want the routes added if recording is disabled. To fix `test_starts_disabled`, couldn't you instead make sure that remote recording is enabled (e.g. with `APPMAP=true`?

Reply:
This was before the rebase of the events pr, so let me try putting it back to see what I get.

Reply:
It was failing two other testcases, not the one I said:
```
FAILED appmap/test/test_flask.py::TestRecordRequestsFlask::test_record_request_appmap_not_enabled_requests_enabled_and_remote - AssertionError
FAILED appmap/test/test_flask.py::TestRecordRequestsFlask::test_record_request_appmap_not_enabled_requests_enabled_no_remote - KeyError: 'appmap-file-name'
```

Also, it was failing this testcases only for Flask.  Apparently there was no such check for the routes for Django.

I think it's because I was trying to test the case of `APPMAP=false` (appmap not enabled) but then enabling remote recording through the testcase issuing a POST.

My understanding is that remote recording and `APPMAP=true` are separate features.  Is this true, or is remote recording possible only if `APPMAP=true`?

Reply:
> Is this true, or is remote recording possible only if APPMAP=true?
Remote recording is only possible if APPMAP=true.

Reply:
> Apparently there was no such check for the routes for Django.

This seems like a bug.

Reply:
I think something's wrong.  In [test_can_record](https://github.com/getappmap/appmap-python/blob/master/appmap/test/web_framework.py#L263) I'm seeing remote requests enabled with a POST but then by the time the DELETE is issued to disable them recording is already off.  As if some `.stop_recording()` call disabled recording.